src/script/event/NotificationService.ts

```diff
@@ -169,4 +168,34 @@ export class NotificationService {
       value: notificationId,
     });
   }
+
+  async getNotificationIdByMessageId(messageId: string, clientId: string): Promise<string | void> {
+    let message: EventRecord;
+
+    if (this.storageService.isTemporaryAndNonPersistent) {
+      message = Object.values(this.storageService.objectDb.events).filter(event => event.id === messageId)[0];
+    } else {
+      message = await this.storageService.db
+        .table(StorageSchemata.OBJECT_STORE.EVENTS)
+        .where({id: messageId})
+        .first();
+    }
+
+    const {notifications} = await this.getNotifications(clientId);
+
+    for (const notification of notifications) {
+      const matchedEvent = notification.payload.find(event => {
```

Comment:
I think this won't work, because every loop iteration will overwrite `matchedEvent`.

Reply:
Whoops, that's still inside the loop, so all good!